kornia/augmentation/_2d/geometric/rotation.py

```diff
@@ -110,3 +110,103 @@ def inverse_transform(
+        slow as it operates based on affine transformations.
+
+    Examples:
+        >>> rng = torch.manual_seed(1)
+        >>> torch.set_printoptions(sci_mode=False)
+        >>> input = torch.tensor([[1., 0., 0., 2.],
+        ...                       [0., 0., 0., 0.],
+        ...                       [0., 1., 2., 0.],
+        ...                       [0., 0., 1., 2.]])
+        >>> aug = RandomRotation90(times=(1, 1), p=1.)
+        >>> out = aug(input)
+        >>> out
+        tensor([[[[    2.0000,     0.0000,     0.0000,     2.0000],
+                  [    0.0000,     0.0000,     2.0000,     1.0000],
+                  [    0.0000,     0.0000,     1.0000,     0.0000],
+                  [    1.0000,     0.0000,     0.0000,     0.0000]]]])
+        >>> aug.transform_matrix
+        tensor([[[    -0.0000,      1.0000,      0.0000],
+                 [    -1.0000,     -0.0000,      3.0000],
+                 [     0.0000,      0.0000,      1.0000]]])
+        >>> inv = aug.inverse(out)
+
+    To apply the exact augmenation again, you may take the advantage of the previous parameter state:
+        >>> input = torch.randn(1, 3, 32, 32)
+        >>> aug = RandomRotation90(times=(-1, 1), p=1.)
+        >>> (aug(input) == aug(input, params=aug._params)).all()
+        tensor(True)
+    """
+
+    def __init__(
+        self,
+        times: Tuple[int, int],
+        resample: Union[str, int, Resample] = Resample.BILINEAR.name,
+        same_on_batch: bool = False,
+        align_corners: bool = True,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        self._param_generator = rg.PlainUniformGenerator((times, "times", 0.0, (-3, 3)))
+
+        self.flags = {"resample": Resample.get(resample), "align_corners": align_corners}
+
+    def compute_transformation(self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]) -> Tensor:
+        # TODO: Update to use `get_rotation_matrix2d`
+        angles: Tensor = 90.0 * params["times"].round().to(input)
+
+        center: Tensor = _compute_tensor_center(input)
+        rotation_mat: Tensor = _compute_rotation_matrix(angles, center.expand(angles.shape[0], -1))
+
+        # rotation_mat is B x 2 x 3 and we need a B x 3 x 3 matrix
+        trans_mat: Tensor = eye_like(3, input, shared_memory=False)
+        trans_mat[:, 0] = rotation_mat[:, 0]
+        trans_mat[:, 1] = rotation_mat[:, 1]
+
+        return trans_mat
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        if not isinstance(transform, Tensor):
+            raise TypeError(f"Expected the `transform` be a Tensor. Got {type(transform)}.")
+
+        return affine(input, transform[..., :2, :3], flags["resample"].name.lower(), "zeros", flags["align_corners"])
```

Comment:
Just a guess but probably rot90 should faster than interpolating?

Reply:
definitely. I only push a workable version here.